diff -up util-linux-2.37/configure.ac.kzak util-linux-2.37/configure.ac
--- util-linux-2.37/configure.ac.kzak	2021-06-01 09:50:39.898388178 +0200
+++ util-linux-2.37/configure.ac	2021-07-14 17:57:45.927112572 +0200
@@ -1452,14 +1452,6 @@ UL_REQUIRES_HAVE([setpriv], [linux_capab
 UL_REQUIRES_HAVE([setpriv], [cap_ng], [libcap-ng library])
 AM_CONDITIONAL([BUILD_SETPRIV], [test "x$build_setpriv" = xyes])
 
-PKG_CHECK_MODULES([PCRE2_POSIX], [libpcre2-posix],
-		  [have_pcre2_posix=yes],
-		  [have_pcre2_posix=no])
-AS_IF([test "x$have_pcre2_posix" = xyes ], [
-    AC_DEFINE([HAVE_PCRE2_POSIX], [1], [Define if libpcre2-posix is available])
-])
-AM_CONDITIONAL([HAVE_PCRE2_POSIX], [test "x$have_pcre2_posix" = xyes])
-
 AC_ARG_ENABLE([hardlink],
   AS_HELP_STRING([--disable-hardlink], [do not build hardlink]),
   [], [UL_DEFAULT_ENABLE([hardlink], [check])]
diff -up util-linux-2.37/misc-utils/hardlink.c.kzak util-linux-2.37/misc-utils/hardlink.c
--- util-linux-2.37/misc-utils/hardlink.c.kzak	2021-05-24 13:39:52.714045529 +0200
+++ util-linux-2.37/misc-utils/hardlink.c	2021-07-14 17:57:51.910085213 +0200
@@ -44,14 +44,7 @@
 #include "monotonic.h"
 #include "optutils.h"
 
-/* Use libpcre2posix if it's available */
-#ifdef HAVE_PCRE2_POSIX
-# include <pcre2posix.h>
-# undef REG_NOSUB
-# define REG_NOSUB 0		/* we do want backreferences in PCRE mode */
-#else
-# include <regex.h>		/* regcomp(), regsearch() */
-#endif
+#include <regex.h>		/* regcomp(), regsearch() */
 
 #ifdef HAVE_SYS_XATTR_H
 # include <sys/xattr.h>		/* listxattr, getxattr */
diff -up util-linux-2.37/misc-utils/Makemodule.am.kzak util-linux-2.37/misc-utils/Makemodule.am
--- util-linux-2.37/misc-utils/Makemodule.am.kzak	2021-05-24 13:39:52.714045529 +0200
+++ util-linux-2.37/misc-utils/Makemodule.am	2021-07-14 17:57:51.910085213 +0200
@@ -244,8 +244,4 @@ dist_noinst_DATA += misc-utils/hardlink.
 hardlink_SOURCES = misc-utils/hardlink.c lib/monotonic.c
 hardlink_LDADD = $(LDADD) libcommon.la $(REALTIME_LIBS)
 hardlink_CFLAGS = $(AM_CFLAGS)
-if HAVE_PCRE2_POSIX
-hardlink_LDADD += $(PCRE2_POSIX_LIBS)
-hardlink_CFLAGS += $(PCRE2_POSIX_CFLAGS)
-endif
 endif
