diff -up util-linux-2.27.1/libblkid/src/superblocks/minix.c.kzak util-linux-2.27.1/libblkid/src/superblocks/minix.c
--- util-linux-2.27.1/libblkid/src/superblocks/minix.c.kzak	2016-01-18 15:02:05.493738910 +0100
+++ util-linux-2.27.1/libblkid/src/superblocks/minix.c	2016-01-18 15:04:20.020773281 +0100
@@ -90,7 +90,8 @@ static int probe_minix(blkid_probe pr, c
 		struct minix_super_block *sb = (struct minix_super_block *) data;
 		int zones, ninodes, imaps, zmaps, firstz;
 
-		if (sb->s_imap_blocks == 0 || sb->s_zmap_blocks == 0)
+		if (sb->s_imap_blocks == 0 || sb->s_zmap_blocks == 0 ||
+		    sb->s_log_zone_size != 0)
 			return 1;
 
 		zones = version == 2 ? minix_swab32(swabme, sb->s_zones) :
