--- util-linux-2.12a/login-utils/login.c.pamstart	2004-09-15 16:13:36.462028734 +0200
+++ util-linux-2.12a/login-utils/login.c	2004-09-15 16:13:50.790278325 +0200
@@ -549,7 +549,7 @@
      * Therefore, we are safe not setting it to anything
      */
 
-    retcode = pam_start("login",username, &conv, &pamh);
+    retcode = pam_start(hflag?"remote":"login",username, &conv, &pamh);
     if(retcode != PAM_SUCCESS) {
 	fprintf(stderr, _("login: PAM Failure, aborting: %s\n"),
 		pam_strerror(pamh, retcode));
