--- util-linux-2.11y/login-utils/chsh.c.sopwith	Fri Sep  5 08:51:49 2003
+++ util-linux-2.11y/login-utils/chsh.c	Fri Sep  5 08:53:43 2003
@@ -185,7 +185,6 @@
 	printf (_("Shell not changed.\n"));
 	return 0;
     }
-    if (!strcmp(shell, "/bin/sh")) shell = "";
     pw->pw_shell = shell;
     if (setpwnam (pw) < 0) {
 	perror ("setpwnam");
@@ -299,6 +298,8 @@
 check_shell (char *shell) {
     int i, c;
 
+    if (! *shell) shell = "/bin/sh"; /* Fudge it */
+
     if (*shell != '/') {
 	printf (_("%s: shell must be a full path name.\n"), whoami);
 	return (-1);
