--- util-linux-2.13-pre7/mount/mount_constants.h.kzak	2007-08-08 16:48:29.000000000 +0200
+++ util-linux-2.13-pre7/mount/mount_constants.h	2007-08-08 16:48:30.000000000 +0200
@@ -57,6 +57,10 @@
 #ifndef MS_VERBOSE
 #define MS_VERBOSE	0x8000	/* 32768 */
 #endif
+#ifndef MS_RELATIME
+#define MS_RELATIME	0x200000 /* 200000: Update access times relative
+                                  to mtime/ctime */
+#endif
 #ifndef MS_UNBINDABLE
 #define MS_UNBINDABLE	(1<<17)	/* 131072 unbindable*/
 #endif
--- util-linux-2.13-pre7/mount/mount.8.kzak	2007-08-08 16:48:29.000000000 +0200
+++ util-linux-2.13-pre7/mount/mount.8	2007-08-08 16:48:30.000000000 +0200
@@ -623,6 +623,13 @@
 .B nodiratime
 Do not update directory inode access times on this filesystem.
 .TP
+.B relatime
+Update inode access times relative to modify or change time.  Access
+time is only updated if the previous access time was earlier than the
+current modify or change time. (Similar to noatime, but doesn't break
+mutt or other applications that need to know if a file has been read
+since the last time it was modified.)
+.TP
 .B noauto
 Can only be mounted explicitly (i.e., the
 .B \-a
--- util-linux-2.13-pre7/mount/mount.c.kzak	2007-08-08 16:48:30.000000000 +0200
+++ util-linux-2.13-pre7/mount/mount.c	2007-08-08 16:53:02.000000000 +0200
@@ -177,6 +177,12 @@
   { "diratime",	0, 1, MS_NODIRATIME },	/* Update dir access times */
   { "nodiratime", 0, 0, MS_NODIRATIME },/* Do not update dir access times */
 #endif
+#ifdef MS_RELATIME
+  { "relatime",	0, 0, MS_RELATIME },   /* Update access times relative to
+						mtime/ctime */
+  { "norelatime", 0, 1, MS_RELATIME }, /* Update access time without regard
+					to mtime/ctime */
+#endif
   { "kudzu", 0, 0, MS_COMMENT },			/* Silently remove this option (backwards compat use only - deprecated) */
   { "managed", 0, 0, MS_COMMENT },			/* Silently remove this option */
   { NULL,	0, 0, 0		}
