--- util-linux-2.13-pre6/mount/umount.c.sysfs	2006-02-23 10:09:53.000000000 +0100
+++ util-linux-2.13-pre6/mount/umount.c	2006-02-23 10:10:39.000000000 +0100
@@ -734,7 +734,7 @@
 	if (all) {
 		/* nodev stuff: sysfs, usbfs, oprofilefs, ... */
 		if (types == NULL)
-			types = "noproc,nodevfs,nodevpts";
+			types = "noproc,nodevfs,nodevpts,nosysfs";
 		result = umount_all (types, test_opts);
 	} else if (argc < 1) {
 		usage (stderr, 2);
