--- util-linux-2.12p/login-utils/login.c.login-lastlog	2005-03-24 13:26:06.516865128 +0100
+++ util-linux-2.12p/login-utils/login.c	2005-03-24 13:26:58.136017824 +0100
@@ -1397,7 +1397,7 @@
     struct lastlog ll;
     int fd;
     
-    if ((fd = open(_PATH_LASTLOG, O_RDWR, 0)) >= 0) {
+    if ((fd = open(_PATH_LASTLOG, O_RDWR|O_CREAT, 0)) >= 0) {
 	lseek(fd, (off_t)pwd->pw_uid * sizeof(ll), SEEK_SET);
 	if (!quiet) {
 	    if (read(fd, (char *)&ll, sizeof(ll)) == sizeof(ll) &&
