--- util-linux-2.12p/login-utils/agetty.c.execl	2005-07-12 15:53:25.000000000 +0200
+++ util-linux-2.12p/login-utils/agetty.c	2005-07-12 15:53:49.000000000 +0200
@@ -382,7 +382,7 @@
 
     /* Let the login program take care of password validation. */
 
-    (void) execl(options.login, options.login, "--", logname, (char *) 0);
+    (void) execl(options.login, options.login, "--", logname, NULL);
     error(_("%s: can't exec %s: %m"), options.tty, options.login);
     exit(0);  /* quiet GCC */
 }
--- util-linux-2.12p/misc-utils/script.c.execl	2005-07-12 15:54:17.000000000 +0200
+++ util-linux-2.12p/misc-utils/script.c	2005-07-12 15:54:28.000000000 +0200
@@ -325,9 +325,9 @@
 		shname = shell;
 
 	if (cflg)
-		execl(shell, shname, "-c", cflg, 0);
+		execl(shell, shname, "-c", cflg, NULL);
 	else
-		execl(shell, shname, "-i", 0);
+		execl(shell, shname, "-i", NULL);
 
 	perror(shell);
 	fail();
