--- util-linux-2.13-pre6/mount/mount.c.sloppy	2006-09-15 08:39:53.000000000 +0200
+++ util-linux-2.13-pre6/mount/mount.c	2006-09-15 08:43:24.000000000 +0200
@@ -554,7 +554,7 @@
        if (stat(mountprog, &statbuf) == 0) {
 	    res = fork();
 	    if (res == 0) {
-		 char *oo, *mountargs[10];
+		 char *oo, *mountargs[11];
 		 int i = 0;
 
 		 setuid(getuid());
@@ -563,6 +563,8 @@
 		 mountargs[i++] = mountprog;
 		 mountargs[i++] = spec;
 		 mountargs[i++] = node;
+		 if (sloppy && (strcmp(type, "nfs")==0 || strcmp(type, "nfs4")==0))
+		      mountargs[i++] = "-s";
 		 if (nomtab)
 		      mountargs[i++] = "-n";
 		 if (verbose)
